### ISSUE TITLE
Proposal: Checkpointing only for `Executor`s internal state and solver

### ISSUE THREAD

**Issue by stefan-k** [argmin-rs/argmin#34]
Checkpointing currently serializes all of the `Executor`, which requires the `ArgminOp` to be serializable (even if no checkpointing is used). However, one actually only needs to serialize the `state` and the `solver` field, the other things can be supplied when starting from a loaded checkpoint.

So instead of 

```rust
let res = Executor::from_checkpoint(".checkpoints/landweber_exec.arg")
        .unwrap_or(Executor::new(operator, solver, init_param))
        .max_iters(iters)
        .checkpoint_dir(".checkpoints")
        .checkpoint_name("landweber_exec")
        .checkpoint_mode(CheckpointMode::Every(20))
        .add_observer(ArgminSlogLogger::term(), ObserverMode::Always)
        .run()?;
```

One could write this like this:

```rust
let res = Executor::from_checkpoint(".checkpoints/landweber_exec.arg", &operator)
        .unwrap_or(Executor::new(operator, solver, init_param))
        .max_iters(iters)
        .checkpoint_dir(".checkpoints")
        .checkpoint_name("landweber_exec")
        .checkpoint_mode(CheckpointMode::Every(20))
        .add_observer(ArgminSlogLogger::term(), ObserverMode::Always)
        .run()?;
```

This would, to a large extent, solve #7 .

Caveat:

* Parameters still need to be serializable (No way around that I guess)
* Operators are not allowed to have internal state which changes over the course of the optimization because it will not be checkpointed (not sure if this really is an issue)

Since solvers may have internal state, they need to be part of the checkpointing.

I'm open to suggestions.

**Comment by stefan-k** [comment]
I started working on it in #36 but unfortunately ran into a problem which I don't really understand ;)

**Comment by stefan-k** [comment]
I have implemented it as outlined above in #36.
